Samples/src/SimpleGrover/SimpleGrover.qs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+namespace Microsoft.Quantum.Samples.SimpleGrover {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Arithmetic;
+
+    // This operation adds a (-1)-phase to the marked state(s)
+    operation ReflectMarked (inputQubits : Qubit[]) : Unit is Adj + Ctl {
+        using (outputQubit = Qubit()) {
+            within {
+                // Initialize the outputQubit to 1/sqrt(2) ( |0> - |1> )
+                // so that toggling it results in a (-1)-phase
+                X(outputQubit);
+                H(outputQubit);
+            }
+            apply {
+                // Flip the outputQubit for marked states.
+                // Here: For the state with alternating 0s and 1s
+                within {
```

Comment:
Would it make sense to remove a level of nesting here?

Reply:
The outer level turns the bit-flip oracle into a phase-flip oracle, and the inner within-apply is the bit-flip oracle itself, so I left the two levels for illustration. Should I put an additional note there or similar?